contracts/governance/src/collection.rs

```diff
@@ -0,0 +1,42 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use alloc::vec::Vec;
+use canonical::CanonError;
+use canonical_derive::Canon;
+
+#[derive(Clone, Canon, Debug)]
+pub struct Collection<K, V> {
+    data: Vec<(K, V)>,
+}
+
+impl<K: PartialEq, V> Collection<K, V> {
+    pub fn get(&self, key: &K) -> Result<Option<&V>, CanonError> {
+        Ok(self.data.iter().find(|(x, _)| x == key).map(|(_, y)| y))
+    }
+
+    pub fn insert(&mut self, key: K, value: V) -> Result<(), CanonError> {
+        self.data.push((key, value));
+
+        Ok(())
+    }
```

Comment:
A map cannot have duplicated keys. We should:

- find the key in the vector and update its corresponding value
- otherwise, push the key/value pair

Reply:
That makes the insert O(n). Hopefully we aren't sacrificing performance here.

Reply:
This is however a mandatory property. This just emulates a map, but won't extend the performance properties of a map (its just a vec)